### PR TITLE
Test/2022 UI api plans run e2e

### PR DESCRIPTION
**Pull request by kkibbey1996**
## Issue

https://gravitee.atlassian.net/browse/APIM-2022 

## Description

Plans page tests for APIs in APIM. 

Splitting out my General and Plan commits, with updated code and faker details. 

I will squash commits etc. after review comments & changes. 

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-audrnpyuji.chromatic.com)

### CODE REVIEW

**Comment by phiz71** [comment]
@kkibbey1996 I don't think you made 113 commits 😅. Could you please rebase your branch to keep only the commits you've done for this PR? 🙏🏻

File: gravitee-apim-e2e/ui-test/integration/apim/ui/apis/ui-api-plans.spec.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    cy.contains('Rate Limiting').should('be.visible');
+    cy.contains('Quota').should('be.visible');
+    cy.contains('Resource Filtering').should('be.visible');
+    cy.get('[type="submit"]').contains('Create').click();
+    cy.contains('Configuration successfully saved!').should('be.visible');
+    cy.get('[type="button"]').contains('STAGING').click();
+    cy.contains(`${planName}-APIKey`).should('be.visible');
+    cy.getByDataTestId('api_plans_closeplan_button').first().click();
+    cy.get(`[placeholder="${planName}-APIKey"]`).type(`${planName}-APIKey`);
+    // cy.contains('Yes, delete this plan').click()
```
Comment (julian-robotham-gravitee) [inline gravitee-apim-e2e/ui-test/integration/apim/ui/apis/ui-api-plans.spec.ts]: Is this comment still needed?

File: gravitee-apim-console-webui/src/management/api/general/plans/list/api-portal-plan-list.component.html [hunk 2]
```diff
@@ -107,27 +108,57 @@
       <div class="plans__table__actions">
         <ng-container *ngIf="!isReadOnly; else readOnlyPlanActions">
           <ng-container *ngIf="element.status !== 'CLOSED'">
-            <button mat-icon-button aria-label="Edit the plan" matTooltip="Edit the plan" (click)="navigateToPlan(element.id)">
+            <button
+              mat-icon-button
+              aria-label="Edit the plan"
+              matTooltip="Edit the plan"
+              data-testid="api_plans_editplan_button"
```
Comment (julian-robotham-gravitee) [inline gravitee-apim-console-webui/src/management/api/general/plans/list/api-portal-plan-list.component.html]: For the reference name and value I would of suggested  `data-test-id="api_plans_edit_plan_button"`
but this is probably a minor point and shouldn't block the PR

**Review by julian-robotham-gravitee — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/general/plans/list/api-portal-plan-list.component.html (hunk 2 above)
Comment (Okhelifi) [reply gravitee-apim-console-webui/src/management/api/general/plans/list/api-portal-plan-list.component.html]: At the beginning of the cypress tests developments the `data-testid` usage came from here https://docs.cypress.io/guides/references/best-practices#How-It-Works I think 🤷🏼 Maybe I'm wrong

**Review by Okhelifi — CHANGES_REQUESTED** [review]
You have an issue with your last rebase I think.

**Pull request closed without merging**